src/history/HistoryArchiveReportWork.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+// Copyright 2020 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "history/HistoryArchive.h"
+#include "work/Work.h"
+
+namespace stellar
+{
+class GetHistoryArchiveStateWork;
+class HistoryArchiveReportWork : public Work
+{
+  protected:
+    BasicWork::State doWork() override;
+    void onSuccess() override;
+    void onFailureRaise() override;
```

Comment:
You also need to implement `doReset` to reset the state in case there are retries (I know we don't retry, but it that changes in the future, this will not work properly). For example, see how it's done in `BatchWork::doReset()`. The rule is thumb is generally like this: if this work is going to retry, which member vars do we need to reset in order for the work to correctly run again?  So here you'd want to reset children works, for example.

Reply:
Got it. I added `doReset` that clears `mGetHistoryArchiveStateWorks`